src/tests/banners/bannerSelection.test.ts

```diff
@@ -0,0 +1,244 @@
+import { selectBannerTest, _ } from './bannerSelection';
+import { BannerTargeting, BannerPageTracking } from '../../components/BannerTypes';
+import { cacheAsync as _cacheAsync } from '../../lib/cache';
+
+const cacheAsync = _cacheAsync;
+
+jest.mock('../../lib/cache', () => ({
+    cacheAsync: jest.fn(),
+}));
+
+describe('selectBannerTest', () => {
+    afterEach(() => {
+        jest.resetAllMocks();
+    });
+
+    const firstDate = 'Mon Jun 06 2020 19:20:10 GMT+0100';
+    const secondDate = 'Mon Jul 06 2020 19:20:10 GMT+0100';
+
+    describe('Australia Moment', () => {
+        let targeting: BannerTargeting;
+        let tracking: BannerPageTracking;
+
+        beforeEach(() => {
+            targeting = {
+                alreadyVisitedCount: 3,
+                shouldHideReaderRevenue: false,
+                isPaidContent: false,
+                showSupportMessaging: true,
+                mvtId: 3,
+                countryCode: 'AU',
+                engagementBannerLastClosedAt: firstDate,
+            };
+            tracking = {
+                ophanPageId: '',
+                ophanComponentId: '',
+                platformId: '',
+                referrerUrl: '',
+                clientName: '',
+            };
+        });
+
+        it('returns banner if it has never been dismissed', () => {
+            // // @ts-ignore
+            (cacheAsync as jest.Mock).mockReturnValue([
+                null,
+                (): Promise<Date> => Promise.resolve(new Date(secondDate)),
+            ]);
+
+            targeting.engagementBannerLastClosedAt = undefined;
+
+            _.resetCache('contributions', 'australia');
+
+            return selectBannerTest(targeting, tracking, '').then(result => {
+                expect(result && result.test.name).toBe('AusMomentContributionsBanner');
+            });
+        });
+
+        it('returns banner if has been redeployed', () => {
+            // // @ts-ignore
+            (cacheAsync as jest.Mock).mockReturnValue([
+                null,
+                (): Promise<Date> => Promise.resolve(new Date(secondDate)),
+            ]);
+
+            _.resetCache('contributions', 'australia');
+
+            return selectBannerTest(targeting, tracking, '').then(result => {
+                expect(result && result.test.name).toBe('AusMomentContributionsBanner');
+            });
+        });
+
+        it('returns null if article is paid content', () => {
+            // // @ts-ignore
+            (cacheAsync as jest.Mock).mockReturnValue([
+                null,
+                (): Promise<Date> => Promise.resolve(new Date(secondDate)),
+            ]);
+
+            _.resetCache('contributions', 'australia');
+
+            targeting.isPaidContent = true;
+
+            return selectBannerTest(targeting, tracking, '').then(result => {
+                expect(result).toBe(null);
+            });
+        });
+    });
+
+    describe('Subs Banner', () => {
```

Comment:
We could also apply the changed mentioned previously in relation to `describe('Australia Moment', () => { ... }` to this `describe('Subs Banner', () => { ... }` block too.

Reply:
Yes, good point